plugin/storage/es/factory.go

```diff
@@ -218,7 +218,7 @@ func (f *Factory) CreateArchiveSpanReader() (spanstore.Reader, error) {
 	if !f.archiveConfig.Enabled {
 		return nil, nil
 	}
-	sr, err := createSpanReader(f.getArchiveClient, f.archiveConfig, true, f.logger, f.tracer)
```

Comment:
it's fine to pass archive flag here, since you're in the archive "context" due to the method name. The objective is not to pass `archive` down to the actual span reader.